types/weixin-app/index.d.ts

```diff
@@ -1,2133 +1,3361 @@
+    /** 是否返回节点的 scrollLeft scrollTop ，节点必须是scroll-view或者viewport */
+    scrollOffset?: true;
+    /**
+     * 指定属性名列表
+     * 返回节点对应属性名的当前属性值（只能获得组件文档中标注的常规属性值， id class style 和事件绑定的属性值不可获取
+     */
+    properties?: string[];
+  }
+  interface NodeRefFieldsValue {
+    id: {
+      id: boolean;
+    };
+    dataset: {
+      dataset: string;
+    };
+    rect: {
+      left: number;
+      right: number;
+      top: number;
+      bottom: number;
+    };
+    size: {
+      width: number;
+      height: number;
+    };
+    scrollOffset: {
+      scrollLeft: number;
+      scrollTop: number;
+    };
+    properties: {
+      properties: any;
+    };
+  }
+  interface NodesRef {
+    /**
+     * 添加节点的布局位置的查询请求，相对于显示区域，以像素为单位。
+     * 其功能类似于DOM的getBoundingClientRect。
+     * 返回值是nodesRef对应的selectorQuery。
+     * 返回的节点信息中，每个节点的位置用
+     * left、right、top、bottom、width、height字段描述。
+     * 如果提供了callback回调函数，在执行selectQuery的exec方法后
+     * 节点信息会在callback中返回。
+     */
+    boundingClientRect<T extends NodesRefRect | NodesRefRect[]>(callback?: (rect: T) => void): SelectorQuery;
+    /**
+     * 添加节点的滚动位置查询请求，以像素为单位。
+     * 节点必须是scroll-view或者viewport。
+     * 返回值是nodesRef对应的selectorQuery。
+     * 返回的节点信息中，每个节点的滚动位置用scrollLeft、scrollHeight字段描述。
+     * 如果提供了callback回调函数，在执行selectQuery的exec方法后，节点信息会在callback中返回。
+     */
+    scrollOffset(callback?: (rect: NodeRefOffset) => void): SelectorQuery;
+    /**
+     * 获取节点的相关信息，需要获取的字段在fields中指定。
+     * 返回值是nodesRef对应的selectorQuery。
+     */
+    fields(fields: NodeRefFieldsOptions, callback?: (result: any) => void): SelectorQuery;
+  }
+  /**
+   * SelectorQuery对象实例
+   */
+  interface SelectorQuery {
+    /**
+     * 将选择器的选取范围更改为自定义组件component内
+     * （初始时，选择器仅选取页面范围的节点，不会选取任何自定义组件中的节点
+     * @version 1.6.0
+     */
+    in(component: Component<object>): SelectorQuery;
+    /**
+     * 在当前页面下选择第一个匹配选择器selector的节点，返回一个NodesRef对象实例，可以用于获取节点信息。
+     * selector类似于CSS的选择器，但仅支持下列语法。
+     * + ID选择器：#the-id
+     * + class选择器（可以连续指定多个）：.a-class.another-class
+     * + 子元素选择器：.the-parent > .the-child
+     * + 后代选择器：.the-ancestor .the-descendant
+     * + 跨自定义组件的后代选择器：.the-ancestor >>> .the-descendant
+     * + 多选择器的并集：#a-node, .some-other-nodes
+     */
+    select(selector: string): NodesRef;
+    /**
+     * 在当前页面下选择匹配选择器selector的节点，返回一个NodesRef对象实例。
+     * 与selectorQuery.selectNode(selector)不同的是，它选择所有匹配选择器的节点。
+     */
+    selectAll(selector: string): NodesRef;
+    /**
+     * 选择显示区域，可用于获取显示区域的尺寸、滚动位置等信息
+     * 返回一个NodesRef对象实例。
+     */
+    selectViewport(): NodesRef;
+    /**
+     * 执行所有的请求
+     * 请求结果按请求次序构成数组，在callback的第一个参数中返回。
+     */
+    exec(callback?: (result: any[]) => void): void;
+  }
+  // #endregion
+  // 界面-----绘图
+  interface CanvasAction {
+    method: string;
+    data: CanvasAction[] | Array<number | string>;
+  }
+  type LineCapType = 'butt' | 'round' | 'square';
+  type LineJoinType = 'bevel' | 'round' | 'miter';
+  /**
+   * context只是一个记录方法调用的容器，用于生成记录绘制行为的actions数组。contex
```

Comment:
Woah, does this basically have the same component API as Vue? In that case, I don't think you need the `V` type parameter since that was just for backwards compatibility. CC @HerringtonDarkholme @ktsn.

Reply:
Indeed! We did keep the `V` type parameter for compatibility. Also, @AlexStacker you might find [this](https://herringtondarkholme.github.io/2017/10/12/vue-ts3/) helpful.

Reply:
@DanielRosenwasser I now understand. I was concerned about the vue 2.5 of the types's PR, so  when I update weixin app this type of file, immediately remembered the vue types file, so directly copied over did not change.